sdcm/cluster.py

```diff
@@ -2205,6 +2205,19 @@ def install_scylla(self, scylla_repo):
                 'sudo apt-get install -y -o Dpkg::Options::="--force-overwrite" -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" '
                 '--allow-unauthenticated {}'.format(self.scylla_pkg()))
 
+        # THIS IS A WORKAROUND FOR ISSUE https://github.com/scylladb/scylla/issues/10442
+        # the issue is related to JDK version, and the fix was added to later patches of multiple base versions,
+        # hence this is a temporary workaround to make the rolling upgrade tests to pass, until the latest
+        # patch of the supported releases will include the fix.
+        package_manager = 'yum' if self.is_rhel_like() else 'apt'
```

Comment:
maybe we should first check which java version is installed as default ? before we apply this patch/hack ?

Reply:
this specific trick is to make 2021.1 upgrade tests to pass, as all the base versions will either not receive the fix (as they are already deprecated), or the fix didn't yet reach a released patch...
we can do it for `master` or other branches, and in that case, we do need to check the installed version first, as we do have a fix already to most versions, so it is not needed